### PR TITLE
OSD-7065: Pull image by digest

### PR DESCRIPTION
**Pull request by 2uasimojo**
- Generate the SelectorSyncSet to reference `${REPO_DIGEST}` as the image pull path. This value is generated by app-sre.
- Refactor the appsre pipeline scripts to skip the build/push if the image already exists. Explanation for this weirdness inline.

[OSD-7065](https://issues.redhat.com/browse/OSD-7065)

### CODE REVIEW

**Comment by jharrington22** [comment]
@lisa will let you unhold but the build pieces reflect what we've done everywhere else for other operators.

/lgtm

File: build/selectorsyncset.yaml [hunk 1]
```diff
@@ -126,8 +126,8 @@ objects:
               - -cacert
               - /service-ca/service-ca.crt
               - -tls
-              image: quay.io/app-sre/managed-cluster-validating-webhooks:${IMAGE_TAG}
-              imagePullPolicy: Always
+              image: ${REPO_DIGEST}
+              imagePullPolicy: IfNotPresent
```
Comment (jharrington22) [inline build/selectorsyncset.yaml:130]: Nice change for digests.

**Comment by lisa** [comment]
> @lisa will let you unhold but the build pieces reflect what we've done everywhere else for other image builds.
> 
> /lgtm

@jharrington22 yes, the build parts look good. I've asked Eric (outside Github) to update the testing section in README.md to include a note about `REPO_DIGEST`.

**Comment by jharrington22** [comment]
@2uasimojo is there a matching SAAS change?

**Comment by jharrington22** [comment]
> > @lisa will let you unhold but the build pieces reflect what we've done everywhere else for other image builds.
> > /lgtm
> 
> @jharrington22 yes, the build parts look good. I've asked Eric (outside Github) to update the testing section in README.md to include a note about `REPO_DIGEST`.

Yeap I saw that, just acking the technical change, I'll leave the documentation formalities to you. If there are a lot of changes I'd prefer to open a OSD card and assign it to Eric to change at a later date (like next week). We have a few images to get through before tomorrow.

**Comment by 2uasimojo** [comment]
> @2uasimojo is there a matching SAAS change already?

Glad you asked. There is no SaaS change for this because we don't need one. These aren't registry images and don't have a SaaS bundle repo or channels; so the OLM template wasn't using a channel in the tag originally; so we don't need to add `use_channel_in_image_tag` in the app-interface SaaS config to make this work (in fact doing so would break it).

**Comment by 2uasimojo** [comment]
@lisa Doc updates are done.

**Review by lisa — APPROVED** [review]
/lgtm
/approve

File: build/build_push.sh [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+# failure.
+# If the query fails for any reason, prints an error and *exits* nonzero.
+#
+# This function cribbed from:
+# https://github.com/openshift/boilerplate/blob/d2d6640ef28a7ce54ac639db32c825aa4bb492a0/boilerplate/_lib/common.sh#L77-L125
+image_exists_in_repo() {
+    local image_uri=$1
+    local output
+
+    output=$(skopeo inspect docker://${image_uri} 2>&1)
```
Comment (2uasimojo) [inline build/build_push.sh:23]: Need to fix this per https://github.com/openshift/boilerplate/pull/162

Comment (2uasimojo) [reply build/build_push.sh:23]: #154

**Pull request merged**